### ISSUE TITLE
Diagnostic error: ReferenceError: Can't find variable: window

### ISSUE THREAD

**Issue by CochiorasBogdan** [Nozbe/WatermelonDB#514]
Hello, with the latest version of WatermleonDB 0.15.0-2 It seems that the web (React) part gives errors at runtime (`npm start` from `create-react-app` project) from the `LokiJS` adaptor:
```
Diagnostic error: ReferenceError: Can't find variable: window
Unhandled Promise Rejection: DataCloneError: The object can not be cloned.
```
It seems to be caused by the latest merged changes from `LokiJS` and more exactly line `294` from: https://github.com/techfort/LokiJS/blob/13ff214b46d99961cfa39dc03f5595a8d205793e/src/incremental-indexeddb-adapter.js

` var openRequest = indexedDB.open("IncrementalAdapterIDB", 1); `

I think `indexedDB` is expected to be on the `window` variable which is `undefined` in service workers and if the `window` variable is mocked as an empty object the indexedDB is `undefined`. Not sure if I am doing something wrong or if anything needs to be updated, with 0.14 there are no issues.

**Comment by radex** [comment]
@CochiorasBogdan thanks for the catch! 0.15.0-2 is an internal beta version - you shouldn't use it until it's released :)

**Comment by radex** [comment]
@CochiorasBogdan let me know if it's fixed on 0.15.0-3
